knowledge_graph/classifier/large_language_model.py

```diff
@@ -164,6 +165,17 @@ def __setstate__(self, state):
 
     def predict(self, text: str) -> list[Span]:
         """Predict whether the supplied text contains an instance of the concept."""
+
+        # Check whether an event loop is already running. Python can't run nested
+        # async processes, so if there's a running loop then we use `nest_asyncio` to
+        # make the prediction be able to run.
+        try:
+            loop = asyncio.get_running_loop()
+            if loop.is_running():
+                nest_asyncio.apply()
+        except RuntimeError:
+            pass
```

Comment:
sticking an escape hatch here smells suboptimal to me...

i know that this isn't the main point of the PR, and i'd guess that it's really just included to unblock the rest, but i think it's worth figuring out why these issues are happening. for what it's worth, i considered a get-out like this when i spotted similar issues around the wikibasesession in prefect, which makes me thing this is a larger problem which we should solve more generally.

maybe there's a common decorator which we could add to methods like this? it could sit alongside the `async_to_sync` stuff in `wikibase.py`? or maybe in a new `knowledge_graph/async.py`?